scripts/drive_paged_programs.py

```diff
@@ -442,8 +454,24 @@ def __metric_calculator(r: torch.Tensor, t: torch.Tensor):
 failed_cases = []
 # for each program and valid prompt (batch size, sequence length)
 for program_id, valid_prompt in valid_prompts:
+    # when we enforce homogeneous prompt programs, we will cycle through all sizes between the min of a program and the valid prompt sequence length
+    # if there does not exist enough sequenece sizes between this range, we will cycle back to the beginning
+    # in the event we don't have enough sequences that satisfy the enforce_sizes, we will repeat sequences and warn the user
+    enforce_sizes = [valid_prompt[1]]
+    if args.enforce_homogeneous_prompt_programs:
+        tkv_cutoff = max_tkv
+        tkv_cutoff = tkv_cutoff // 2
+        while tkv_cutoff >= valid_prompt[1]:
+            tkv_cutoff = tkv_cutoff // 2
```

Comment:
this will end up with a tkv_cutoff that's smaller than valid_prompt[1]. Is that ok?

Reply:
and why are we starting from max_tkv if the range is from tkv_cutoff to vaslid_prompt[1]?